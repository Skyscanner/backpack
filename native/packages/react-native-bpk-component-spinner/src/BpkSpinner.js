/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow */

import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator } from 'react-native';
import {
  getThemeAttributes,
  makeThemePropType,
  withTheme,
} from 'react-native-bpk-theming';
import {
  colorBlue500,
  colorWhite,
  colorGray700,
} from 'bpk-tokens/tokens/base.react.native';

const SPINNER_TYPES = {
  primary: 'primary',
  light: 'light',
  dark: 'dark',
};

type SpinnerType = $Keys<typeof SPINNER_TYPES>;

const REQUIRED_THEME_ATTRIBUTES: Array<string> = ['spinnerPrimaryColor'];

const getSpinnerColor = (themeAttributes: ?Object, type: SpinnerType) => {
  const colorMappings = {
    [SPINNER_TYPES.primary]: colorBlue500,
    [SPINNER_TYPES.light]: colorWhite,
    [SPINNER_TYPES.dark]: colorGray700,
  };

  if (themeAttributes && type === 'primary') {
    return themeAttributes.spinnerPrimaryColor;
  }

  return colorMappings[type];
};

const BpkSpinner = props => {
  const { small, type, ...rest } = props;
  const { theme } = props;
  let themeAttributes = getThemeAttributes(REQUIRED_THEME_ATTRIBUTES, theme);

  // Validate type.
  if (!Object.keys(SPINNER_TYPES).includes(type)) {
    throw new Error(
      `"${type}" is not a valid spinner type. Valid types are ${Object.keys(
        SPINNER_TYPES,
      ).join(', ')}`,
    );
  }

  // Validate that spinner is themeable and correct theming attribute has been
  // supplied. If not, disable theming.
  if (themeAttributes && type !== 'primary') {
    themeAttributes = null;
  }

  return (
    // eslint-disable-next-line backpack/use-components
    <ActivityIndicator
      color={getSpinnerColor(themeAttributes, type)}
      size={small ? 'small' : 'large'}
      {...rest}
    />
  );
};

const propTypes = {
  small: PropTypes.bool,
  theme: makeThemePropType(REQUIRED_THEME_ATTRIBUTES),
  type: PropTypes.oneOf(Object.keys(SPINNER_TYPES)),
};

BpkSpinner.propTypes = propTypes;

BpkSpinner.defaultProps = {
  small: false,
  theme: null,
  type: 'primary',
};

export default withTheme(BpkSpinner);
export { propTypes, SPINNER_TYPES };
