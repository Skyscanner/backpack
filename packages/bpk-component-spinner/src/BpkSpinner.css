@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-spinner{width:1.5rem;height:1.5rem;animation:bpk-keyframe-spin 600ms linear infinite}.bpk-spinner--primary{fill:#0770e3;fill:var(--bpk-spinner-primary-color, #0770e3)}.bpk-spinner--light{fill:#fff}.bpk-spinner--dark{fill:#444560}.bpk-spinner--large{width:2rem;height:2rem}.bpk-spinner--extra-large{width:3rem;height:3rem}.bpk-spinner--align-to-button{vertical-align:text-bottom}.bpk-spinner--align-to-large-button{margin-top:.5rem;vertical-align:top}
