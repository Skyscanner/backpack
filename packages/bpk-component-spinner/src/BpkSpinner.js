/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import PropTypes from 'prop-types';
import React from 'react';
import SmSpinner from '@skyscanner/bpk-svgs/dist/js/spinners/sm';

import { cssModules } from '../../bpk-react-utils';

import STYLES from './BpkSpinner.module.scss';
import SPINNER_TYPES from './spinnerTypes';

const getClassName = cssModules(STYLES);

type Props = {
  type: $Keys<typeof SPINNER_TYPES>,
  className: ?string,
  alignToButton: boolean,
};

const BpkSpinner = (props: Props) => {
  const { alignToButton, className, type, ...rest } = props;

  const classNames = getClassName(
    'bpk-spinner',
    `bpk-spinner--${type}`,
    alignToButton && 'bpk-spinner--align-to-button',
    className,
  );

  // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
  return <SmSpinner className={classNames} {...rest} />;
};

BpkSpinner.propTypes = {
  type: PropTypes.oneOf(Object.keys(SPINNER_TYPES)),
  className: PropTypes.string,
  alignToButton: PropTypes.bool,
};

BpkSpinner.defaultProps = {
  type: SPINNER_TYPES.dark,
  className: null,
  alignToButton: false,
};

export default BpkSpinner;
