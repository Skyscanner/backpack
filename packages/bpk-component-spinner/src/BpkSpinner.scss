/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2019 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~bpk-mixins/index';

.bpk-spinner {
  @include bpk-spinner;

  &--primary {
    @include bpk-themeable-property(fill, --bpk-spinner-primary-color, $bpk-color-sky-blue);
  }

  &--light {
    fill: $bpk-color-white;
  }

  &--dark {
    fill: $bpk-color-sky-gray-tint-01;
  }

  &--large {
    @include bpk-spinner--lg;
  }

  &--extra-large {
    @include bpk-spinner--xl;
  }

  &--align-to-button {
    @include bpk-spinner--align-to-button;
  }

  &--align-to-large-button {
    @include bpk-spinner--align-to-large-button;
  }
}
