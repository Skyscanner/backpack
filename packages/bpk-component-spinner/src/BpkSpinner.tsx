/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import SmSpinner from '@skyscanner/bpk-svgs/dist/js/spinners/sm';

import { cssModules } from '../../bpk-react-utils';

import SPINNER_TYPES from './spinnerTypes';

import type { SpinnerTypes } from './spinnerTypes';

import STYLES from './BpkSpinner.module.scss';

const getClassName = cssModules(STYLES);

type Props = {
  type?: SpinnerTypes;
  className?: string | null;
  alignToButton?: boolean;
  [rest: string]: any;
};

const BpkSpinner = ({
  alignToButton = false,
  className = null,
  type = SPINNER_TYPES.dark,
  ...rest
}: Props) => {
  const classNames = getClassName(
    'bpk-spinner',
    `bpk-spinner--${type}`,
    alignToButton && 'bpk-spinner--align-to-button',
    className,
  );

  return (
    <span className={classNames}>
      <SmSpinner {...rest} />
    </span>
  );
};

export default BpkSpinner;
