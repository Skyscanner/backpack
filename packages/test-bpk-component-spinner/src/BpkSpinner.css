@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-spinner{width:1.125rem;height:1.125rem;animation:bpk-keyframe-spin 600ms linear infinite}.bpk-spinner--primary{fill:#0770e3;fill:var(--bpk-spinner-primary-color, #0770e3)}.bpk-spinner--light{fill:#fff}.bpk-spinner--dark{fill:#444560}.bpk-spinner--large{width:1.5rem;height:1.5rem}.bpk-spinner--extra-large{width:2.25rem;height:2.25rem}.bpk-spinner--align-to-button{vertical-align:text-bottom}.bpk-spinner--align-to-large-button{margin-top:.1875rem;vertical-align:top}
