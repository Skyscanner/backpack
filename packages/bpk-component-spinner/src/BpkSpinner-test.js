/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import BpkSpinner from './BpkSpinner';
import SPINNER_TYPES from './spinnerTypes';

describe('BpkSpinner', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkSpinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with type as light', () => {
    const tree = renderer
      .create(<BpkSpinner type={SPINNER_TYPES.light} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with type as primary', () => {
    const tree = renderer
      .create(<BpkSpinner type={SPINNER_TYPES.primary} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with a "className" attribute', () => {
    const tree = renderer
      .create(<BpkSpinner className="my-custom-class" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with a "alignToButton" attribute', () => {
    const tree = renderer.create(<BpkSpinner alignToButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
