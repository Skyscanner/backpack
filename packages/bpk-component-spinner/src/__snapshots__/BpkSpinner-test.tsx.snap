// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkSpinner should render correctly 1`] = `
<DocumentFragment>
  <span
    class="bpk-spinner bpk-spinner--dark"
  >
    <svg
      aria-hidden="true"
      height="1rem"
      viewBox="0 0 16 16"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 7H1a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2"
        opacity="0.45"
      />
      <path
        d="M15 7h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2"
      />
      <path
        d="M7 13v2a1 1 0 1 0 2 0v-2a1 1 0 1 0-2 0"
        opacity="0.3"
      />
      <path
        d="M7 1v2a1 1 0 0 0 2 0V1a1 1 0 0 0-2 0"
        opacity="0.65"
      />
      <path
        d="m9.634 12.83 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 0 0-1.732 1"
        opacity="0.25"
      />
      <path
        d="m3.634 2.438 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 1 0-1.732 1"
        opacity="0.55"
      />
      <path
        d="m11.83 11.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.2"
      />
      <path
        d="m1.438 5.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.5"
      />
      <path
        d="m12.83 6.366 1.732-1a1 1 0 0 0-1-1.732l-1.732 1a1 1 0 0 0 1 1.732"
        opacity="0.85"
      />
      <path
        d="m2.438 12.366 1.732-1a1 1 0 1 0-1-1.732l-1.732 1a1 1 0 1 0 1 1.732"
        opacity="0.4"
      />
      <path
        d="m11.366 4.17 1-1.732a1 1 0 0 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.75"
      />
      <path
        d="m5.366 14.562 1-1.732a1 1 0 1 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.35"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`BpkSpinner should render correctly with a "alignToButton" attribute 1`] = `
<DocumentFragment>
  <span
    class="bpk-spinner bpk-spinner--dark bpk-spinner--align-to-button"
  >
    <svg
      aria-hidden="true"
      height="1rem"
      viewBox="0 0 16 16"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 7H1a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2"
        opacity="0.45"
      />
      <path
        d="M15 7h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2"
      />
      <path
        d="M7 13v2a1 1 0 1 0 2 0v-2a1 1 0 1 0-2 0"
        opacity="0.3"
      />
      <path
        d="M7 1v2a1 1 0 0 0 2 0V1a1 1 0 0 0-2 0"
        opacity="0.65"
      />
      <path
        d="m9.634 12.83 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 0 0-1.732 1"
        opacity="0.25"
      />
      <path
        d="m3.634 2.438 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 1 0-1.732 1"
        opacity="0.55"
      />
      <path
        d="m11.83 11.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.2"
      />
      <path
        d="m1.438 5.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.5"
      />
      <path
        d="m12.83 6.366 1.732-1a1 1 0 0 0-1-1.732l-1.732 1a1 1 0 0 0 1 1.732"
        opacity="0.85"
      />
      <path
        d="m2.438 12.366 1.732-1a1 1 0 1 0-1-1.732l-1.732 1a1 1 0 1 0 1 1.732"
        opacity="0.4"
      />
      <path
        d="m11.366 4.17 1-1.732a1 1 0 0 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.75"
      />
      <path
        d="m5.366 14.562 1-1.732a1 1 0 1 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.35"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`BpkSpinner should render correctly with a "className" attribute 1`] = `
<DocumentFragment>
  <span
    class="bpk-spinner bpk-spinner--dark my-custom-class"
  >
    <svg
      aria-hidden="true"
      height="1rem"
      viewBox="0 0 16 16"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 7H1a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2"
        opacity="0.45"
      />
      <path
        d="M15 7h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2"
      />
      <path
        d="M7 13v2a1 1 0 1 0 2 0v-2a1 1 0 1 0-2 0"
        opacity="0.3"
      />
      <path
        d="M7 1v2a1 1 0 0 0 2 0V1a1 1 0 0 0-2 0"
        opacity="0.65"
      />
      <path
        d="m9.634 12.83 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 0 0-1.732 1"
        opacity="0.25"
      />
      <path
        d="m3.634 2.438 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 1 0-1.732 1"
        opacity="0.55"
      />
      <path
        d="m11.83 11.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.2"
      />
      <path
        d="m1.438 5.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.5"
      />
      <path
        d="m12.83 6.366 1.732-1a1 1 0 0 0-1-1.732l-1.732 1a1 1 0 0 0 1 1.732"
        opacity="0.85"
      />
      <path
        d="m2.438 12.366 1.732-1a1 1 0 1 0-1-1.732l-1.732 1a1 1 0 1 0 1 1.732"
        opacity="0.4"
      />
      <path
        d="m11.366 4.17 1-1.732a1 1 0 0 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.75"
      />
      <path
        d="m5.366 14.562 1-1.732a1 1 0 1 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.35"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`BpkSpinner should render correctly with type as light 1`] = `
<DocumentFragment>
  <span
    class="bpk-spinner bpk-spinner--light"
  >
    <svg
      aria-hidden="true"
      height="1rem"
      viewBox="0 0 16 16"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 7H1a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2"
        opacity="0.45"
      />
      <path
        d="M15 7h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2"
      />
      <path
        d="M7 13v2a1 1 0 1 0 2 0v-2a1 1 0 1 0-2 0"
        opacity="0.3"
      />
      <path
        d="M7 1v2a1 1 0 0 0 2 0V1a1 1 0 0 0-2 0"
        opacity="0.65"
      />
      <path
        d="m9.634 12.83 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 0 0-1.732 1"
        opacity="0.25"
      />
      <path
        d="m3.634 2.438 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 1 0-1.732 1"
        opacity="0.55"
      />
      <path
        d="m11.83 11.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.2"
      />
      <path
        d="m1.438 5.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.5"
      />
      <path
        d="m12.83 6.366 1.732-1a1 1 0 0 0-1-1.732l-1.732 1a1 1 0 0 0 1 1.732"
        opacity="0.85"
      />
      <path
        d="m2.438 12.366 1.732-1a1 1 0 1 0-1-1.732l-1.732 1a1 1 0 1 0 1 1.732"
        opacity="0.4"
      />
      <path
        d="m11.366 4.17 1-1.732a1 1 0 0 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.75"
      />
      <path
        d="m5.366 14.562 1-1.732a1 1 0 1 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.35"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`BpkSpinner should render correctly with type as primary 1`] = `
<DocumentFragment>
  <span
    class="bpk-spinner bpk-spinner--primary"
  >
    <svg
      aria-hidden="true"
      height="1rem"
      viewBox="0 0 16 16"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 7H1a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2"
        opacity="0.45"
      />
      <path
        d="M15 7h-2a1 1 0 1 0 0 2h2a1 1 0 1 0 0-2"
      />
      <path
        d="M7 13v2a1 1 0 1 0 2 0v-2a1 1 0 1 0-2 0"
        opacity="0.3"
      />
      <path
        d="M7 1v2a1 1 0 0 0 2 0V1a1 1 0 0 0-2 0"
        opacity="0.65"
      />
      <path
        d="m9.634 12.83 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 0 0-1.732 1"
        opacity="0.25"
      />
      <path
        d="m3.634 2.438 1 1.732a1 1 0 1 0 1.732-1l-1-1.732a1 1 0 1 0-1.732 1"
        opacity="0.55"
      />
      <path
        d="m11.83 11.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.2"
      />
      <path
        d="m1.438 5.366 1.732 1a1 1 0 1 0 1-1.732l-1.732-1a1 1 0 1 0-1 1.732"
        opacity="0.5"
      />
      <path
        d="m12.83 6.366 1.732-1a1 1 0 0 0-1-1.732l-1.732 1a1 1 0 0 0 1 1.732"
        opacity="0.85"
      />
      <path
        d="m2.438 12.366 1.732-1a1 1 0 1 0-1-1.732l-1.732 1a1 1 0 1 0 1 1.732"
        opacity="0.4"
      />
      <path
        d="m11.366 4.17 1-1.732a1 1 0 0 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.75"
      />
      <path
        d="m5.366 14.562 1-1.732a1 1 0 1 0-1.732-1l-1 1.732a1 1 0 0 0 1.732 1"
        opacity="0.35"
      />
    </svg>
  </span>
</DocumentFragment>
`;
